pkg/snapshot/snapshotter/garbagecollector.go

```diff
@@ -193,24 +195,35 @@ func getSnapStreamIndexList(snapList brtypes.SnapList) []int {
 	return snapStreamIndexList
 }
 
-// garbageCollectChunks deletes the chunks in the store from snaplist starting at index low (inclusive) till high (exclusive).
-func garbageCollectChunks(store brtypes.SnapStore, snapList brtypes.SnapList, low, high int) {
-	for index := low; index < high; index++ {
-		snap := snapList[index]
-		// Only delete chunk snapshots of kind Full
-		if snap.Kind != brtypes.SnapshotKindFull || !snap.IsChunk {
+// GarbageCollectChunks removes obsolete chunks based on the latest recorded snapshot.
+// It eliminates chunks associated with snapshots that have already been uploaded.
+// Additionally, it avoids deleting chunks linked to snapshots currently being uploaded to prevent the garbage collector from removing chunks before the composite is formed.
+func (ssr *Snapshotter) GarbageCollectChunks(snapList brtypes.SnapList) (int, brtypes.SnapList) {
```

Comment:
Why you are making this func as public ?

Reply:
That was a requirement for the unit tests. I was writing unit tests so I had to make it accessible for other packages. For eg, GC is in snapshotter pkg but unit tests for GC functions are in snapshotter_test pkg, so to use the func for unit tests, I had to make those it public. 

All other funcs are also public for the exact same reason. So I changed this func also as it was necessary.